src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -877,8 +893,54 @@ const put = async (frameworkName, config, rawConfig) => {
     throw createError('Forbidden', 'ForbiddenUserError', `User ${userName} is not allowed to do operation in ${virtualCluster}`);
   }
 
-  const storageConfig = await userModel.getUserStorageConfigs(userName);
-  const frameworkDescription = generateFrameworkDescription(frameworkName, virtualCluster, config, rawConfig, storageConfig);
+  // check deprecated storages config
+  if (
+    'extras' in config &&
+    !config.extras.storages &&
+    'com.microsoft.pai.runtimeplugin' in config.extras
+  ) {
+    for (let plugin of config.extras['com.microsoft.pai.runtimeplugin']) {
+      if (plugin.plugin === 'teamwise_storage') {
+        if ('parameters' in plugin && plugin.parameters.storageConfigNames) {
+          config.extras.storages =
+            plugin.parameters.storageConfigNames.map((name) => {
+              return {name};
+            });
+        } else {
+          config.extras.storages = [];
+        }
+      }
+    }
+  }
```

Comment:
Can we delete the `teamwise_storage` in config?
If `config.extras.storages` is set. Just delete the `teamwise_storage` plugin.
If `config.extras.storages` is not set, but `teamwise_storage` is here, convert to `extras.storage` and delete the `teamwise_stirage` in config.
May need to notice user the config is changed.

In this way, we will decouple the `rest-server` and `runtime` for storage part.

Reply:
1. it's not recommended to modify user's config
2. rest server is not aware of `com.microsoft.pai.runtimeplugin` field, which is validated, parsed and used in runtime

The current solution is just a workaround for backward compatibility, modifying this config needs more complex validation and parsing for `com.microsoft.pai.runtimeplugin` field.

To decouple storage part, why not just delete the storage plugin code in runtime simply.
If those code may be reused in the future, just rename the plugin name and there won't exist conflicts.

Reply:
That's make sense.